### PR TITLE
lgc: Pass Layer and ViewportIndex from VS to HS/GS and from HS to DS

### PR DESCRIPTION
**Pull request by xazhangAMD**
DX spec allows SV_RenderTargetArrayIndex/SV_ViewportArrayIndex to be used in those shader stages.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
0df5f9a4699775cb5f8696280a542ab8b4f7648f Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:34: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
                                  ^

**Comment by amdvlk-admin** [comment]
fcdbe00ed6993f00fccb585f00d543c8b7c0aa93 Jenkins build error.
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/builder/InOutBuilder.cpp:1560:16: error: ‘struct lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed>’ has no member named ‘viewportIndexIn’; did you mean ‘viewportIndex’?
 1560 |       usage.gs.viewportIndexIn = true;
      |                ^~~~~~~~~~~~~~~
      |                viewportIndex
[2936/3149] Building CXX object compiler/llpc/llvm/tools/lgc/CMakeFiles/LLVMlgc.dir/builder/ImageBuilder.cpp.o
FAILED: compiler/llpc/llvm/tools/lgc/CMakeFiles/LLVMlgc.dir/builder/ImageBuilder.cpp.o 
--
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:267:34: error: redeclaration of ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  267 |         unsigned viewportIndex : 1;        // Whether gl_ViewportIndex is used
      |                                  ^
/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llpc/lgc/include/lgc/state/ResourceUsage.h:259:18: note: previous declaration ‘unsigned int lgc::ResourceUsage::<unnamed struct>::<unnamed union>::<unnamed struct>::viewportIndex’
  259 |         unsigned viewportIndex : 1;  // Whether DX SV_ViewportArrayIndex is used
      |                  ^~~~~~~~~~~~~

File: lgc/patch/PatchInOutImportExport.cpp [hunk 1]
```diff
@@ -2183,6 +2183,14 @@ Value *PatchInOutImportExport::patchTcsBuiltInInputImport(Type *inputTy, unsigne
...
     break;
   }
+  case BuiltInLayer:
+  case BuiltInViewportIndex: {
+    assert(builtInInLocMap.find(builtInId) != builtInInLocMap.end());
+    const unsigned loc = builtInInLocMap.find(builtInId)->second;
+    auto ldsOffset = calcLdsOffsetForTcsInput(inputTy, loc, nullptr, elemIdx, vertexIdx, builder);
+    input = readValueFromLds(false, inputTy, ldsOffset, builder);
+    break;
+  }
```
Comment (nhaehnle) [inline lgc/patch/PatchInOutImportExport.cpp:2193]: I believe their handling should be identical to BuiltInPointSize. Please just add the additional `case` statements there.

File: lgc/patch/PatchInOutImportExport.cpp [hunk 2]
```diff
@@ -2309,6 +2317,20 @@ Value *PatchInOutImportExport::patchTesBuiltInInputImport(Type *inputTy, unsigne
...
+    input = readValueFromLds(m_pipelineState->isTessOffChip(), inputTy, ldsOffset, builder);
+    break;
+  }
+  case BuiltInViewportIndex: {
+    assert(builtInInLocMap.find(builtInId) != builtInInLocMap.end());
+    const unsigned loc = builtInInLocMap.find(builtInId)->second;
+    auto ldsOffset = calcLdsOffsetForTesInput(inputTy, loc, nullptr, elemIdx, vertexIdx, builder);
+    input = readValueFromLds(m_pipelineState->isTessOffChip(), inputTy, ldsOffset, builder);
+    break;
+  }
```
Comment (nhaehnle) [inline lgc/patch/PatchInOutImportExport.cpp:2333]: Same here: just add extra `case` statements to BuiltInPointSize.

File: lgc/patch/PatchInOutImportExport.cpp [hunk 3]
```diff
@@ -3086,6 +3130,14 @@ void PatchInOutImportExport::patchTcsBuiltInOutputExport(Value *output, unsigned
 
     break;
   }
+  case BuiltInLayer:
+  case BuiltInViewportIndex: {
```
Comment (nhaehnle) [inline lgc/patch/PatchInOutImportExport.cpp:3134]: Add to the BuiltInPosition / BuiltInPointSize case above.

File: lgc/patch/PatchResourceCollect.cpp [hunk 4]
```diff
@@ -1850,8 +1850,22 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
 
-      builtInUsage.vs.layer = false;
-      builtInUsage.vs.viewportIndex = false;
+      if (nextBuiltInUsage.layerIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInLayer) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInLayer];
+        inOutUsage.builtInOutputLocMap[BuiltInLayer] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.vs.layer = false;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:1859]: We try to follow the rule that if one side of an if/else uses braces, then so should the other. So please add braces here.

File: lgc/patch/PatchResourceCollect.cpp [hunk 5]
```diff
@@ -1850,8 +1850,22 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
+
+      if (nextBuiltInUsage.viewportIndexIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInViewportIndex) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInViewportIndex];
+        inOutUsage.builtInOutputLocMap[BuiltInViewportIndex] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.vs.viewportIndex = false;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:1867]: Same here: add braces

File: lgc/patch/PatchResourceCollect.cpp [hunk 6]
```diff
@@ -1890,8 +1904,22 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
 
-      builtInUsage.vs.layer = false;
-      builtInUsage.vs.viewportIndex = false;
+      if (nextBuiltInUsage.layerIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInLayer) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInLayer];
+        inOutUsage.builtInOutputLocMap[BuiltInLayer] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.vs.layer = 0;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:1913]: Same here: add braces

File: lgc/patch/PatchResourceCollect.cpp [hunk 7]
```diff
@@ -1890,8 +1904,22 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
+
+      if (nextBuiltInUsage.viewportIndexIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInViewportIndex) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInViewportIndex];
+        inOutUsage.builtInOutputLocMap[BuiltInViewportIndex] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.vs.viewportIndex = 0;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:1921]: Same here: add bracesa

File: lgc/patch/PatchResourceCollect.cpp [hunk 8]
```diff
@@ -2186,8 +2246,21 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
 
-      builtInUsage.tes.layer = false;
-      builtInUsage.tes.viewportIndex = false;
+      if (nextBuiltInUsage.layerIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInLayer) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInLayer];
+        inOutUsage.builtInOutputLocMap[BuiltInLayer] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.tes.layer = 0;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:2255]: Same here: add braces

File: lgc/patch/PatchResourceCollect.cpp [hunk 9]
```diff
@@ -2186,8 +2246,21 @@ void PatchResourceCollect::mapBuiltInToGenericInOut() {
...
+
+      if (nextBuiltInUsage.viewportIndexIn) {
+        assert(nextInOutUsage.builtInInputLocMap.find(BuiltInViewportIndex) != nextInOutUsage.builtInInputLocMap.end());
+        const unsigned mapLoc = nextInOutUsage.builtInInputLocMap[BuiltInViewportIndex];
+        inOutUsage.builtInOutputLocMap[BuiltInViewportIndex] = mapLoc;
+        availOutMapLoc = std::max(availOutMapLoc, mapLoc + 1);
+      } else
+        builtInUsage.tes.viewportIndex = 0;
```
Comment (nhaehnle) [inline lgc/patch/PatchResourceCollect.cpp:2263]: Same here: add braces

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8b1c8bb1dcc332f30fd59fb84317e864fd06d0e7
<details><summary>CTS tests (Failed: 0/139021)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69517 (59.2%)</li>
<li>Failed: 0/69517 (0.0%)</li>
<li>Not Supported: 28373/69517 (40.8%)</li><li>Warnings: 0/69517 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28326/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchInOutImportExport.cpp (hunk 2 above)
Comment (nhaehnle) [reply lgc/patch/PatchInOutImportExport.cpp:2327]: This one can still be converted to just being extra `case` statements, right?

**Review by nhaehnle — COMMENTED** [review]
Thanks, but I think you missed a spot.

**Comment by xazhangAMD** [comment]
> Thanks, but I think you missed a spot.

Yeah, I missed that. Will update soon.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0bcbcd90253a9e3f2f34c5d04596c19d1e89ad34
<details><summary>CTS tests (Failed: 0/139021)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69517 (59.2%)</li>
<li>Failed: 0/69517 (0.0%)</li>
<li>Not Supported: 28373/69517 (40.8%)</li><li>Warnings: 0/69517 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28326/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7cb1a160ea1c687a195037057403bd30df09f6e3
<details><summary>CTS tests (Failed: 0/139021)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69517 (59.2%)</li>
<li>Failed: 0/69517 (0.0%)</li>
<li>Not Supported: 28373/69517 (40.8%)</li><li>Warnings: 0/69517 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41177/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28327/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**